src/page_header.h

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+
+#ifndef LIBSQL_PAGE_HEADER_H
+#define LIBSQL_PAGE_HEADER_H
+
+typedef struct sqlite3_pcache_page sqlite3_pcache_page;
+typedef struct Pager Pager;
```

Comment:
Any chance you could find a way to keep these types with a libsql or sqlite3 prefix? Essentially everything else in the amalgam header is prefixed.

Reply:
These particular structs are meant to stay opaque though. Virtual WAL implementers should have access to page number and its data (in order to be able to implement xFrames), but not to internal sqlite page cache structures. Ideally, the public part should live in another struct, but that complicates backward compatibility

Reply:
(that said, I don't mind typedefing them to have a customary prefix, and it should definitely be done for struct Wal, which gets exposed as public API in libsql)

Reply:
Ah, if these aren't part of the `sqlite3.h`/`sqlite3ext.h` header then it doesn't matter.

Reply:
It's a little more complicated, they are... `PgHdr` is an internal struct, but it contains two fields virtual WAL needs - `pData` and `pgno`. At the same time, all of its fields are used internally. So, as a sad result, the struct is exposed in a public header with a few opaque type definitions. These opaque type declarations are harmless, but ugly. I don't love this solution, but I'm undecided how best to solve it. An alternative would be to prepend all the opaque type names during amalgamation generation with a `libsql_` prefix, or maintain a second header which mimics the `PgHdr` structure exactly, but with `void *` pointers instead of the internal types.

If anyone has a better suggestion, I'd love to hear it!

/cc @penberg